src/Mollie/WC/Gateway/Abstract.php

```diff
@@ -2136,4 +2141,104 @@ protected function appendOrderArgumentsToUrl($order_id, $order_key, $webhook_url
+        );
+
+        // Overwrite gateway-wide
+        $new_order_status = apply_filters(
+            Mollie_WC_Plugin::PLUGIN_ID . $statusTypeString . $this->id,
+            $new_order_status
+        );
+        return $new_order_status;
+    }
+
+    /**
+     * @param Mollie\Api\Resources\Payment|Mollie\Api\Resources\Order $payment
+     *
+     * @return bool
+     */
+    protected function isPartialRefund($payment)
+    {
+        return ($payment->amount->value - $payment->amountRefunded->value) != 0;
+    }
+
+    /**
+     * @param WC_Order                                                $order
+     * @param Mollie\Api\Resources\Payment|Mollie\Api\Resources\Order $payment
+     */
+    protected function processUpdateStateRefund(WC_Order $order, $payment)
+    {
+        if ($this->isPartialRefund($payment)) {
+            $this->updateStateRefund(
+                $order,
+                $payment,
+                self::STATUS_ON_HOLD,
+                '_order_status_partially_refunded',
+                'partially refunded'
+            );
+            return;
+        }
+        $this->updateStateRefund(
+            $order,
+            $payment,
+            self::STATUS_REFUNDED,
+            '_order_status_refunded',
+            'refunded'
+        );
+    }
+
+    /**
+     * @param WC_Order                                                $order
+     * @param Mollie\Api\Resources\Payment|Mollie\Api\Resources\Order $payment
+     * @param                                                         $new_order_status
+     * @param                                                         $statusFilterString
+     * @param                                                         $statusNoteString
+     */
+    protected function updateStateRefund(
+        WC_Order $order,
+        $payment,
+        $new_order_status,
+        $statusFilterString,
+        $statusNoteString
+    ) {
+        // New order status
+        $new_order_status = $this->overwriteFilters(
+            $new_order_status,
+            $statusFilterString
+        );
+        $payment_method_title = $this->getPaymentMethodTitle($payment);
+
+        // Update order status for order with charged_back payment, don't restore stock
+        $this->updateOrderStatus(
+            $order,
+            $new_order_status,
+            sprintf(
+            /* translators: Placeholder 1: payment method title, placeholder 2: payment ID */
+                __(
+                    '%s payment %s via Mollie (%s). You will need to manually review the payment (and adjust product stocks if you use it).',
+                    'mollie-payments-for-woocommerce'
+                ),
+                $payment_method_title,
+                $statusNoteString,
+                $payment->id . ($payment->mode == 'test' ? (' - ' . __(
+                        'test mode',
+                        'mollie-payments-for-woocommerce'
+                    )) : '')
+            ),
```

Comment:
I would put the sprintf in a separated method because there's a lot of stuffs. 
Secondary I'd prevent to add 

```
$payment->id . ($payment->mode == 'test' ? (' - ' . __(
                        'test mode',
                        'mollie-payments-for-woocommerce'
                    )) : '')
```

In one line of code an preferred variables assignments.